src/Tasks/Al.cs

```diff
@@ -305,7 +305,12 @@ protected override string GenerateFullPathToTool()
 
             if (String.IsNullOrEmpty(pathToTool) || !FileSystems.Default.FileExists(pathToTool))
             {
-                pathToTool = SdkToolsPathUtility.GeneratePathToTool(SdkToolsPathUtility.FileInfoExists, ProcessorArchitecture.CurrentProcessArchitecture, SdkToolsPath, ToolExe, Log, true);
+                // The bitness of al.exe should match the platform being built
+                string archToLookFor =  Platform.Equals("x86", StringComparison.OrdinalIgnoreCase) ? Platform :
```

Comment:
The task has a `Prefer32Bit` parameters which also plays a role in the tool invocation. It is intentional to ignore it here and use only `Platform`, is that correct?

Reply:
I think that's only if the platform is AnyCPU.

Reply:
> The task has a Prefer32Bit parameters which also plays a role in the tool invocation. It is intentional to ignore it here and use only Platform, is that correct?

I was sort of torn here. Prefer32Bit has existed here for a while, but I decided _not_ to use it for the sake of the customer. I haven't seen a single customer _want_ to use x64 Al on a 32 bit platform (and vise-versa), and I'm not sure a customer would want that warning? If we take on `Prefer32Bit` (which `csc` uses), we increase complexity unless it's some new AlPrefer32Bit which is a new opt in for customers. 

If we just use Platform customer's don't need to do anything and it "just works."